static/js/unread.js

```diff
@@ -29,6 +33,28 @@ export function set_messages_read_in_narrow(value) {
 
 export const unread_mentions_counter = new Set();
 const unread_messages = new Set();
+// Key is "stream:topic" and values (type Set) are the messages with unread mentions.
+export const unread_mention_topics = new Map();
```

Comment:
We should make the comment clear about how it's computing the keys -- I assume `stream_id` and `topic_name.lower()`? And maybe reference `recent_topics_util.get_topic_key` as the library to access.

We should make sure we handle live-updates that move messages between topics here; but that should be solvable. Worst case, in some circumstances, we could clear this map and refill it using the main data structures.

Reply:
I guess that didn't really answer the main question. I think it's fine to add this data structure to get this feature out; it seems like there might be a refactor that can be done that would avoid needing it, but it seems likely cheap enough to maintain that it'd just be a code cleanup to do so. And I don't see a nice way to avoid it.

Reply:
> We should make sure we handle live-updates that move messages between topics here; but that should be solvable. Worst case, in some circumstances, we could clear this map and refill it using the main data structures.

Ah! I missed this!